### PR TITLE
Nuvoton common module

### PR DESCRIPTION
**Pull request by brentlu**
Create an new common module of Nuvoton amplifier support code for future reuse.

### CODE REVIEW

File: sound/soc/intel/boards/sof_nau8825.c [hunk 1]
```diff
@@ -329,20 +289,6 @@ static struct snd_soc_dai_link_component dmic_component[] = {
 	}
 };
 
-static struct snd_soc_dai_link_component rt1019p_component[] = {
-	{
-		.name = "RTL1019:00",
-		.dai_name = "HiFi",
-	}
-};
```
Comment (plbossart) [inline sound/soc/intel/boards/sof_nau8825.c:337]: this...

File: sound/soc/intel/boards/sof_nau8825.c [hunk 2]
```diff
@@ -478,17 +424,13 @@ sof_card_dai_links_create(struct device *dev, enum sof_ssp_codec amp_type,
...
+			nau8318_set_dai_link(&links[id]);
 			break;
 		case CODEC_RT1015P:
 			sof_rt1015p_dai_link(&links[id]);
 			break;
 		case CODEC_RT1019P:
-			links[id].codecs = rt1019p_component;
-			links[id].num_codecs = ARRAY_SIZE(rt1019p_component);
-			links[id].init = speaker_codec_init;
+			sof_rt1019p_dai_link(&links[id]);
```
Comment (plbossart) [inline sound/soc/intel/boards/sof_nau8825.c:433]: .. and this belong in a different patch IMHO, that's not related to Nuvoton and that's clearly a cleanup we missed earlier.

**Review by plbossart — COMMENTED** [review]
Code looks good @brentlu, I would still split the RT1019p in a different patch .

File: sound/soc/intel/boards/sof_nau8825.c (hunk 1 above)
Comment (brentlu) [reply sound/soc/intel/boards/sof_nau8825.c:337]: done

File: sound/soc/intel/boards/sof_nau8825.c (hunk 2 above)
Comment (brentlu) [reply sound/soc/intel/boards/sof_nau8825.c:433]: done

**Review by plbossart — APPROVED** [review]
Thanks @brentlu,  nice cleanup.

If you have spare cycles, we probably need to update the sof-test remove.sh and insert.sh, the list of modules isn't accurate any longer I think

**Review by bardliao — APPROVED** [review]
